llama-rs/src/inference_session.rs

```diff
@@ -0,0 +1,574 @@
+use std::fmt::Display;
+
+use partial_sort::PartialSort;
+use rand::{distributions::WeightedIndex, prelude::Distribution};
+use thiserror::Error;
+
+use crate::{
+    util::mulf, EvaluateOutputRequest, InferenceError, InferenceParameters, Model, TokenId,
+    TokenUtf8Buffer, EOT_TOKEN_ID,
+};
+
+// The size of a scratch buffer used for inference. This is used for temporary
+// storage of intermediate results during inference.
+//
+// The specific value was copied from `llama.cpp`.
+const SCRATCH_SIZE: usize = 512 * 1024 * 1024;
```

Comment:
I think llama.cpp figured out a proper way to compute this value. We should have a look at this. Not in this PR of course :+1:

Reply:
![image](https://user-images.githubusercontent.com/707827/231720471-efb6e198-3a03-4365-922f-287bca1d23a3.png)

https://github.com/ggerganov/llama.cpp/blob/82d146df9b43cf677e0dbce20b03cf864958a0cc/llama.cpp#L45-L57